product_internal_reference_code/models/res_company.py

```diff
@@ -2,13 +2,57 @@
 # Copyright 2017 Quartile Limited
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from odoo import models, fields
+from odoo import models, fields, api
 
 
 class ResCompany(models.Model):
     _inherit = "res.company"
 
     internal_code_prefix = fields.Char(
         string="Internal Reference Prefix",
-        copy=True,
+        copy=False,
+        require=True,
+        help="A prefix code which is used to automatically generate the "
+             "internal reference when creating a new product."
     )
```

Comment:
I prefer not to have this field.

Reply:
If this field is removed, users have to activate the debug mode to change the prefix directly (Settings > Technical / Sequences & Identifiers / Sequences)

Reply:
I think that's fine as part of initial configuration.  Besides the risk of not making the adjustment in ir.sequence record is negligible.

We want to keep the changes to the system minimal for now as the requirements might later be changed.